.github/workflows/update-completed-sprint-on-issue-closed.yml

```diff
@@ -0,0 +1,127 @@
+name: Update Project Field on Issue/PR Closed
+
+on:
+  workflow_call:
+    inputs:
+      project_name:
+        description: 'Name of the Project which tracks completed sprints'
+        type: 'string'
+        required: true
+      field_name:
+        description: 'Name of the Field in the project to update'
+        type: 'string'
+        required: false
+        default: 'CompletedSprint'
+
+jobs:
+  update_completed_sprint:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Get latest completed sprint
+        id: get_latest_completed_sprint
+        uses: actions/github-script@v5
+        with:
+          script: |
+            const eventName = context.eventName;
+            const url = eventName === 'issues'
+              ? payload.issue.html_url
+              : payload.pull_request.html_url;
+            const { data } = await github.graphql(`
+              query {
+                resource(url: "${url}") {
+                  ... on ${eventName.startsWith('issue') ? 'Issue' : 'PullRequest'} {
+                    id
+                    projectItems(first: 10) {
+                      nodes {
+                        id
+                        project {
+                          id
+                        }
+                      }
+                    }
+                    projectsV2(query: "${inputs.project_name}", first: 1) {
+                      nodes {
+                        id
+                        field(name: "${inputs.field_name}") {
+                          ... on ProjectV2SingleSelectField {
+                            id
+                            options {
+                              id
+                              name
+                            }
+                          }
+                        }
+                      }
+                    }
+                  }
+                }
+              }
+            `);
+
+            const resource = data.resource;
+            if (!data.resource) {
+              throw new Error(`Issue/PR not found: ${url}`);
+            }
+
+            const projects = data.resource.projectsV2.nodes;
+            const projectItems = data.resource.projectItems.nodes?.filter(item =>
+              item.project.id === project.id
+            );
+
+            if (projects.length !== 1 || projectItems.length !== 1) {
+              console.log(`Item ${url} is not a member of the project: ${project_name}`);
+              return
```

Comment:
💡 If multiple projects match the filter the user will see the error saying its not a member of any. Seems like a pretty unlikely edge case though. But would be cheap to add another case in here such as `>==1 { "Item is in multiple projects of that name" }`.